### DISCUSSION TITLE
Using the document field on the front-end

### DISCUSSION THREAD

**Discussion by junaid33** [keystonejs/keystone#7791] [Feature requests]
I'm trying to build a Next.js application that uses the keystone document editor to build pages. If the signed-in user owns the page, they will see the editor and can update the content and save it to the database. If they don’t, they just see the page.
It can be a super simple and extendable website builder. From what I'm reading, it's not possible to to export the document field cleanly.

**Comment by flexdinesh** [comment]
Hey Junaid! That's what we've done in [document-fields-demo](https://keystonejs.com/docs/guides/document-field-demo#document-fields-demo) in the keystone website. The document editor is not built to be exported and used in external applications so you might have to follow the source and do something similar. The source code for what you see in the demo can be found here - https://github.com/keystonejs/keystone/blob/main/docs/components/docs/DocumentEditorDemo.tsx

**Reply by junaid33** [reply]
Thanks, I'll take a look

**Reply by junaid33** [reply]
I'm having trouble importing a lot of the packages.
